packages/extension/src/ui/features/actions/AddNetworkScreen.tsx

```diff
@@ -0,0 +1,191 @@
+
+import { addNetworks } from "../../../background/customNetworks"
+import { Network } from "../../../shared/networks"
+import { BackButton } from "../../components/BackButton"
+import { Button, ButtonGroupVertical } from "../../components/Button"
+import { Header } from "../../components/Header"
+import { InputText } from "../../components/InputText"
+import { FormError, H2 } from "../../components/Typography"
+import { routes } from "../../routes"
+import { isValidAddress } from "../../services/addresses"
+import { TokenDetails } from "../accountTokens/tokens.state"
+
+const AddTokenScreenWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  padding: 0 32px 48px 32px;
+
+  > form {
+    width: 100%;
+    display: flex;
+    flex-direction: column;
+    gap: 16px;
+  }
+
+  ${Button} {
+    margin-top: 64px;
+  }
+`
+
+interface AddNetworkScreenProps {
+  requestedNetwork: Network
+  hideBackButton?: boolean
+  onSubmit?: () => void
+  onReject?: () => void
+}
+
+export const AddNetworkScreen: FC<AddNetworkScreenProps> = ({
+  requestedNetwork,
+  hideBackButton,
+  onSubmit,
+  onReject,
+}) => {
+  const navigate = useNavigate()
+
+  const [error, setError] = useState("")
+
+  //   const validAddress = useMemo(() => {
+  //     return isValidAddress(networkChainId)
+  //   }, [networkChainId])
+
+  //   useEffect(() => {
+  //     if (
+  //       requestedNetwork &&
+  //       requestedNetwork.chainId === networkChainId &&
+  //       !tokenDetails
+  //     ) {
+  //       setLoading(true)
+  //     }
+  //   }, [requestedNetwork, networkChainId, tokenDetails])
+
+  //   useEffect(() => {
+  //     if (account) {
+  //       if (loading && account) {
+  //         fetchTokenDetails(networkChainId, account)
+  //           .then((details) => {
+  //             setTokenDetails(details)
+  //           })
+  //           .catch(() => {
+  //             setTokenDetails(undefined)
+  //           })
+  //           .finally(() => {
+  //             setLoading(false)
+  //           })
+  //       } else if (
+  //         isValidAddress(networkChainId) &&
+  //         networkChainId !== prevValidAddress.current
+  //       ) {
+  //         prevValidAddress.current = networkChainId
+  //         setLoading(true)
+  //       }
+  //     }
+  //   }, [loading, networkChainId, account])
+
+  //   const compiledData = {
+  //     address: networkChainId,
+  //     ...(tokenDetails ?? {}),
+  //     ...(!tokenDetails?.name && { name: networkName }),
+  //     ...(!tokenDetails?.symbol && { symbol: networkBaseUrl }),
+  //     ...(!tokenDetails?.decimals && {
+  //       decimals: BigNumber.from(networkExplorerUrl || "0"),
+  //     }),
+  //     networkId: switcherNetworkId,
+  //   }
+
+  return (
+    <>
+      <Header hide={hideBackButton}>
+        <BackButton />
+      </Header>
+
+      <AddTokenScreenWrapper>
+        <H2>Add Network</H2>
+
+        <form
+          onSubmit={(e: React.FormEvent) => {
```

Comment:
type is inferred here

```suggestion
          onSubmit={(e) => {
```